src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -570,68 +654,18 @@ private async Task RefreshKeyValueCollections(CancellationToken cancellationToke
                             RequestTracingEnabled = _requestTracingEnabled,
                             RequestTracingOptions = _requestTracingOptions
                         },
-                        cancellationToken).ConfigureAwait(false);
-
-                    success = true;
-                }
-                finally
-                {
-                    UpdateCacheExpirationTime(changeWatcher, success);
-                }
+                        cancellationToken)
+                    .ConfigureAwait(false));
 
-                if (keyValueChanges.Any())
-                {
-                    ProcessChanges(keyValueChanges);
-
-                    await SetData(_applicationSettings, false, cancellationToken).ConfigureAwait(false);
-                }
             }
+
+            return keyValueChanges;
         }
 
-        private async Task SetData(IDictionary<string, ConfigurationSetting> data, bool ignoreFailures, CancellationToken cancellationToken)
+        private void SetData(IDictionary<string, string> data, CancellationToken cancellationToken)

```

Comment:
CancellationToken can be removed from this method as its not being used.

Reply:
Thanks for catching this!